ChildProject/pipelines/metrics.py

```diff
@@ -0,0 +1,391 @@
+
+        chi = its[its['speaker_type'] == 'CHI']
+        cries = chi['cries'].apply(lambda x: len(ast.literal_eval(x))).sum()
+        vfxs = chi['vfxs'].apply(lambda x: len(ast.literal_eval(x))).sum()
+        utterances = chi['utterances_count'].sum()
+
+        metrics['lp_n'] = utterances/(utterances + cries + vfxs)
+        metrics['lp_dur'] = chi['utterances_length'].sum()/(chi['child_cry_vfx_len'].sum()+chi['utterances_length'].sum())
+
+        metrics['wc_adu_ph'] = its['words'].sum()*3600/unit_duration
+
+        metrics[self.by] = unit
+        metrics['child_id'] = its['child_id'].iloc[0]
+        metrics['duration'] = unit_duration
+
+        return pd.DataFrame([metrics])
+
+    
+    def extract(self):
+        recordings = self.get_recordings()
+
+        if self.threads == 1:
+            self.metrics = pd.concat([self._process_unit(unit) for unit in recordings[self.by].unique()])
+        else:
+            with mp.Pool(processes = self.threads if self.threads >= 1 else mp.cpu_count()) as pool:
+                self.metrics = pd.concat(pool.map(self._process_unit, recordings[self.by].unique()))
+
+        self.metrics.set_index(self.by, inplace = True)
+        return self.metrics
+        
+
+    @staticmethod
+    def add_parser(subparsers, subcommand):
+        parser = subparsers.add_parser(subcommand, help = 'LENA metrics')
+        parser.add_argument('set', help = 'name of the LENA its annotations set')
+        parser.add_argument('--threads', help = 'amount of threads to run on', default = 1, type = int)
+
+class AclewMetrics(Metrics):
+    """ACLEW metrics extractor.
+    Extracts a number of metrics from the ACLEW pipeline annotations, which includes:
+
+     - The Voice Type Classifier by Lavechin et al. (arXiv:2005.12656)
+     - The Automatic LInguistic Unit Count Estimator (ALICE) by Räsänen et al. (doi:10.3758/s13428-020-01460-x)
+
+    :param project: ChildProject instance of the target dataset.
+    :type project: ChildProject.projects.ChildProject
+    :param set: name of the set associated to the .its annotations
+    :type set: str
+    :param recordings: recordings to sample from; if None, all recordings will be sampled, defaults to None
+    :type recordings: Union[str, List[str], pd.DataFrame], optional
+    :param by: units to sample from, defaults to 'recording_filename'
+    :type by: str, optional
+    :param threads: amount of threads to run on, defaults to 1
+    :type threads: int, optional
+    """
+
+    SUBCOMMAND = 'aclew'
+
+    def __init__(self,
+        project: ChildProject.projects.ChildProject,
+        vtc: str = 'vtc',
+        alice: str = 'alice',
+        vcm: str = 'vcm',
+        recordings: Union[str, List[str], pd.DataFrame] = None,
+        by: str = 'recording_filename',
+        threads: int = 1):
+
+        super().__init__(project, by, recordings)
+
+        self.vtc = vtc
+        self.alice = alice
```

Comment:
have you checked what happens if the user doesn't have an alice analysis? or (this won't typically happen) a vtc analysis?

Reply:
(or a vcm analysis -- which does happen quite often outside of ACLEW data)

Reply:
It will work if VCM and/or ALICE are missing, but the VTC is required.
I could mention it in the docs.
Do you think the VTC should be optional?

Reply:
rather, I think if VCM or ALICE are missing, it would be helpful to give a warning that tells the user that we're not computing those bc they are missing (and if they think they are not missing, to check parameters).
For VTC, instead, it should be an error saying that vtc is not found, and check parameters if they do have it.

Reply:
Done 30f4236